cluster-health-check/cluster-health.sh

```diff
@@ -0,0 +1,519 @@
+  ${KUBECTL_CMD} -n $NAMESPACE get daemonset -o yaml  > $TMPDIR/$NAMESPACE/get-daemonset.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get deployment -o wide  > $TMPDIR/$NAMESPACE/get-deployment.wide 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get deployment -o yaml  > $TMPDIR/$NAMESPACE/get-deployment.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get replicaset -o wide  > $TMPDIR/$NAMESPACE/get-replicaset.wide 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get replicaset -o yaml  > $TMPDIR/$NAMESPACE/get-replicaset.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get job -o wide  > $TMPDIR/$NAMESPACE/get-job.wide 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get job -o yaml  > $TMPDIR/$NAMESPACE/get-job.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get cronjob -o wide  > $TMPDIR/$NAMESPACE/get-cronjob.wide 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get cronjob -o yaml  > $TMPDIR/$NAMESPACE/get-cronjob.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get configmap  > $TMPDIR/$NAMESPACE/get-configmap 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get configmap -o yaml  > $TMPDIR/$NAMESPACE/get-configmap.yaml 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get secret  > $TMPDIR/$NAMESPACE/get-secret 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get pvc -o wide > $TMPDIR/$NAMESPACE/get-pvc.wide 2>&1
+  ${KUBECTL_CMD} -n $NAMESPACE get pvc -o yaml  > $TMPDIR/$NAMESPACE/get-pvc.yaml 2>&1
+}
+storage() {
+  techo "Collecting information about storage..."
+  mkdir -p $TMPDIR/storage/
+  ${KUBECTL_CMD} get storageclass > $TMPDIR/storage/get-storageclass 2>&1
+  ${KUBECTL_CMD} get storageclass -o yaml > $TMPDIR/storage/get-storageclass.yaml 2>&1
+  mkdir -p $TMPDIR/storage/pv/
+  ${KUBECTL_CMD} get pv -o wide > $TMPDIR/storage/get-pv.wide 2>&1
+  ${KUBECTL_CMD} get pv -o yaml > $TMPDIR/storage/get-pv.yaml 2>&1
+}
+archive() {
+
+  FILEDIR=$(dirname $TMPDIR)
+  FILENAME="$(hostname)-$(date +'%Y-%m-%d_%H_%M_%S').tar"
+  tar --create --file ${FILEDIR}/${FILENAME} --directory ${TMPDIR}/ .
+  ## gzip separately for Rancher OS
+  gzip ${FILEDIR}/${FILENAME}
+
+  techo "Created ${FILEDIR}/${FILENAME}.gz"
+
+}
+cleanup() {
+
+  techo "Removing ${TMPDIR}"
+  rm -r -f "${TMPDIR}" >/dev/null 2>&1
+
+}
+help() {
+
+  echo "Cluster Health Check
+  Usage: cluster-health.sh [ -d <directory> -k ~/.kube/config -i rancherlabs/swiss-army-knife -t -c -f -D ]
+
+  All flags are optional
+  -d    Output directory for temporary storage and .tar.gz archive (ex: -d /var/tmp)
+  -k    Override the kubeconfig (ex: ~/.kube/custom)
+  -t    Skip collecting logs and only run tests.
+  -c    Don't cleanup swiss-army-knife test containers
+  -f    Force collection if the minimum space isn't available
+  -i    Override the debug image (ex: registry.example.com/rancherlabs/swiss-army-knife)
+  -D    Enable debug logging"
+
+}
+timestamp() {
+
+  date "+%Y-%m-%d %H:%M:%S"
+
+}
+techo() {
+  echo "$(timestamp): $*"
+}
+decho() {
+  if [[ ! -z $DEBUG ]]
+  then
+    techo "$*"
+  fi
+}
+
+testsonly=0
+cleanup=0
+
+while getopts ":d:s:r:i:tcfhD" opt; do
```

Comment:
Since you already have various steps split out in different functions, is it possible to call a certain function?